### PR TITLE
New actions

### PR DESCRIPTION
**Pull request by zwolf**
Deploy to Staging action uploads to Azure, and runs on merge to master or manual dispatch.
Run Tests action runs on PR open/sync or manual dispatch. Can replace Travis when confirmed working.

### CODE REVIEW

File: .github/workflows/deploy_production.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    - name: Upload to blob storage
+      uses: azure/CLI@v1
+      with:
+        creds: ${{ secrets.AZURE_STATIC_SITES }}
+        inlineScript: |
+            az storage blob upload-batch \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/alice.zooniverse.org' \
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:37]: Still pointed to alice here.
```suggestion
              --destination '$web/translations.zooniverse.org' \
```

File: .github/workflows/deploy_production.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+              --source ./build
+            az storage blob upload \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, max-age=60, s-maxage=60' \
+              --container-name '$web' \
+              --name 'alice.zooniverse.org/index.html' \
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:43]: Same here.
```suggestion
              --name 'translations.zooniverse.org/index.html' \
```

File: .github/workflows/deploy_production.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            "icon_emoji": ":octocat:",
+            "username": "Deploy Action",
+            "attachments": [{
+              "color": '${{ job.status }}' === 'success' ? 'good' : '${{ job.status }}' === 'failure' ? 'danger' : 'warning',
+              "mrkdwn_in": ["text"],
+              "author_name": "${{ github.actor }}",
+              "author_link": "https://github.com/${{ github.actor }}/",
+              "author_icon": "https://github.com/${{ github.actor }}.png?size=40",
+              "title": "ALICE Production deploy complete",
+              "title_link": "https://alice.zooniverse.org",
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:66]:
```suggestion
              "title_link": "https://translations.zooniverse.org",
```

File: .github/workflows/deploy_production.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            "channel": "#deploys",
+            "icon_emoji": ":octocat:",
+            "username": "Deploy Action",
+            "attachments": [{
+              "color": '${{ job.status }}' === 'success' ? 'good' : '${{ job.status }}' === 'failure' ? 'danger' : 'warning',
+              "mrkdwn_in": ["text"],
+              "author_name": "${{ github.actor }}",
+              "author_link": "https://github.com/${{ github.actor }}/",
+              "author_icon": "https://github.com/${{ github.actor }}.png?size=40",
+              "title": "ALICE Production deploy complete",
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:65]:
```suggestion
              "title": "Translations Production deploy complete",
```

File: .github/workflows/deploy_staging.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/pandora.zooniverse.org' \
+              --source ./build
+            az storage blob upload \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, max-age=60, s-maxage=60' \
+              --container-name '$web' \
+              --name 'pandora.zooniverse.org/index.html' \
+              --file ./build/index.html
```
Comment (srallen) [inline .github/workflows/deploy_staging.yml:43]:
```suggestion
              --file ./dist/index.html
```

File: .github/workflows/deploy_staging.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            "icon_emoji": ":octocat:",
+            "username": "Deploy Action",
+            "attachments": [{
+              "color": '${{ job.status }}' === 'success' ? 'good' : '${{ job.status }}' === 'failure' ? 'danger' : 'warning',
+              "mrkdwn_in": ["text"],
+              "author_name": "${{ github.actor }}",
+              "author_link": "https://github.com/${{ github.actor }}/",
+              "author_icon": "https://github.com/${{ github.actor }}.png?size=40",
+              "title": "Pandora Staging deploy complete",
+              "title_link": "https://pandora.preview.zooniverse.org",
```
Comment (srallen) [inline .github/workflows/deploy_staging.yml:65]:
```suggestion
              "title_link": "https://pandora.zooniverse.org",
```

File: .github/workflows/deploy_staging.yml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    - name: Upload to blob storage
+      uses: azure/CLI@v1
+      with:
+        creds: ${{ secrets.AZURE_STATIC_SITES }}
+        inlineScript: |
+            az storage blob upload-batch \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/pandora.zooniverse.org' \
+              --source ./build
```
Comment (srallen) [inline .github/workflows/deploy_staging.yml:37]:
```suggestion
              --source ./dist
```

File: .github/workflows/deploy_production.yml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/alice.zooniverse.org' \
+              --source ./build
+            az storage blob upload \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, max-age=60, s-maxage=60' \
+              --container-name '$web' \
+              --name 'alice.zooniverse.org/index.html' \
+              --file ./build/index.html
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:44]:
```suggestion
              --file ./dist/index.html
```

File: .github/workflows/deploy_production.yml [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    - name: Upload to blob storage
+      uses: azure/CLI@v1
+      with:
+        creds: ${{ secrets.AZURE_STATIC_SITES }}
+        inlineScript: |
+            az storage blob upload-batch \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/alice.zooniverse.org' \
+              --source ./build
```
Comment (srallen) [inline .github/workflows/deploy_production.yml:38]:
```suggestion
              --source ./dist
```

File: .github/workflows/run_tests.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Node.js build
+      id: build
+      uses: actions/setup-node@v1
+      with:
+        node-version: '14.x'
+    - run: npm install
+    - run: npm build
```
Comment (srallen) [inline .github/workflows/run_tests.yml:22]: I don't think this is necessary for test running
```suggestion
```

**Review by srallen — CHANGES_REQUESTED** [review]
I've made a few suggestions, mostly things needing adjustment from the alice versions these borrow from.

File: .github/workflows/deploy_staging.yml (hunk 6 above)
Comment (zwolf) [reply .github/workflows/deploy_staging.yml:65]: `pandora.` is the staging domain, `translations.` is production.

**Comment by zwolf** [comment]
Awesome, thanks :shipit:

File: .github/workflows/deploy_staging.yml (hunk 6 above)
Comment (srallen) [reply .github/workflows/deploy_staging.yml:65]: The original npm script I see deploying staging to `pandora.zooniverse.org`. Is it now `pandora.preview.zooniverse.org` on Azure?

Comment (zwolf) [reply .github/workflows/deploy_staging.yml:65]: Oh, I read this backwards. Yeah you're right

**Review by srallen — APPROVED** [review]

**Pull request merged**